src/Stream.ts

```diff
@@ -61,6 +61,48 @@ export interface Emitter<A> {
   end(): void
 }
 
+export const fromAsyncIterable: {
+  <A, R>(iterable: AsyncGenerator<A, R>): Fx.Fx<Async.Async | Stream<A>, R>
+  <A>(iterable: AsyncIterable<A>): Fx.Fx<Async.Async | Stream<A>, unknown>
+} = <A>(iterable: AsyncIterable<A>): Fx.Fx<Async.Async | Stream<A>, unknown> => Fx.bracket(
+  Fx.sync(() => iterable[Symbol.asyncIterator]()),
+  iterator => Async.run(() => (iterator.return?.() ?? Promise.resolve()).then(() => { })),
+  iterator => Fx.fx(function* () {
+    let next = yield* Async.run(() => iterator.next())
+    while (!next.done) {
+      yield* event(next.value)
+      next = yield* Async.run(() => iterator.next())
+    }
+    return next.value
+  })
+)
+
+export const toAsyncIterable = <Error, Event, A>(fx: Fx.Fx<Async.Async | Fail.Fail<Error> | Stream<Event>, A>): AsyncIterable<Event> => ({
```

Comment:
Do you think it's possible to do something similar here, and return the `A`?  That way the caller can choose to ignore it or not.

Reply:
It's slightly non-standard, but I've added a signature change to capture the return value here - https://github.com/briancavalier/fx/pull/5/commits/8d5bac8640084704a004884cbe382b46112df6e7